menu/src/hooks/useHudListenersService.tsx

```diff
@@ -115,14 +117,32 @@ export const useHudListenersService = () => {
   });
 
   // Handler for dynamically opening the player page & player modal with target
-  useNuiEvent<number>("openPlayerModal", (targetId) => {
-    const targetPlayer = onlinePlayers.find(
-      (playerData) => playerData.id === targetId
-    );
+  useNuiEvent<string>("openPlayerModal", (target) => {
+    let targetPlayer
+    let targetId = parseInt(target)
```

Comment:
Not reassigned anywhere.
```suggestion
    const targetId = parseInt(target)
```